### PR TITLE
Move callstacks from SamplingProfiler to CaptureData

### PR DESCRIPTION
**Pull request by IrinaShkviro**
Callstacks are the "raw" data that we get during the capture, so it
should be part of CaptureData. After this change SamplingProfiler
stores only post-processed data.

### CODE REVIEW

File: OrbitGl/TimeGraph.cpp [hunk 1]
```diff
@@ -646,14 +646,28 @@ std::vector<CallstackEvent> TimeGraph::SelectEvents(float a_WorldStart,
   sampling_profiler->SetGenerateSummary(a_TID == 0);
 
   for (const CallstackEvent& event : selected_callstack_events) {
-    if (Capture::GSamplingProfiler->GetCallStack(event.callstack_hash())) {
-      sampling_profiler->AddCallStack(CallstackEvent(event));
+    uint64_t callstack_id = event.callstack_hash();
+    if (Capture::capture_data_.GetCallstackData()->HasCallStack(callstack_id)) {
```
Comment (dimitry-) [inline OrbitGl/TimeGraph.cpp:650]: introduce a variable for GetCallstackData()/GetSelectionCallstackData() - it will be easier to read code below

File: OrbitCore/SamplingProfiler.cpp [hunk 2]
```diff
@@ -132,7 +115,7 @@ void SamplingProfiler::SortByThreadUsage() {
        });
 }
 
-void SamplingProfiler::ProcessSamples() {
+void SamplingProfiler::ProcessSamples(CallstackData* callstack_data) {
```
[the remark this answers is not in the document]
Comment (IrinaShkviro) [reply OrbitCore/SamplingProfiler.cpp]: Made it const ref here and in ResolveCallstack

File: OrbitCore/CallstackData.h [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  void ForEachUniqueCallstack(
+      const std::function<void(const CallStack&)>& action) {
+    absl::MutexLock lock(&unique_callstacks_mutex_);
+    for (const auto& it : unique_callstacks_) {
+      action(*it.second);
+    }
+  }
+
+  void ForEachFrameInCallstack(uint64_t callstack_id,
+                               const std::function<void(uint64_t)>& action) {
```
[the remark this answers is not in the document]
Comment (IrinaShkviro) [reply OrbitCore/CallstackData.h]: Done

File: OrbitCore/CallstackData.h [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    return unique_callstacks_.at(callstack_id);
+  }
+
+  bool HasCallStack(CallstackID callstack_id) {
+    absl::MutexLock lock(&unique_callstacks_mutex_);
+    return unique_callstacks_.count(callstack_id) > 0;
+  }
+
+  void ForEachUniqueCallstack(
+      const std::function<void(const CallStack&)>& action) {
```
[the remark this answers is not in the document]
Comment (IrinaShkviro) [reply OrbitCore/CallstackData.h]: Done

File: OrbitCore/CallstackData.h [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  [[nodiscard]] uint32_t GetCallstackEventsSize() {
+    return callstack_events_.size();
+  };
+
+  std::shared_ptr<CallStack> GetCallStack(CallstackID callstack_id) {
+    absl::MutexLock lock(&unique_callstacks_mutex_);
+    return unique_callstacks_.at(callstack_id);
+  }
+
+  bool HasCallStack(CallstackID callstack_id) {
```
[the remark this answers is not in the document]
Comment (IrinaShkviro) [reply OrbitCore/CallstackData.h]: Done

File: OrbitCore/CallstackData.h [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+  void AddUniqueCallStack(std::shared_ptr<CallStack> call_stack);
+
+  [[nodiscard]] BlockChain<orbit_client_protos::CallstackEvent, 16 * 1024>&
+  GetCallstackEvents() { return callstack_events_; };
+
+  [[nodiscard]] uint32_t GetCallstackEventsSize() {
+    return callstack_events_.size();
+  };
+
+  std::shared_ptr<CallStack> GetCallStack(CallstackID callstack_id) {
```
[the remark this answers is not in the document]
Comment (IrinaShkviro) [reply OrbitCore/CallstackData.h]: I think that it could be a const reference

File: OrbitCore/CallstackData.h [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+  explicit CallstackData() = default;
+
+  void AddCallstackEvent(orbit_client_protos::CallstackEvent&& callstack_event);
+  void AddUniqueCallStack(CallStack call_stack);
+  void AddUniqueCallStack(std::shared_ptr<CallStack> call_stack);
+
+  [[nodiscard]] BlockChain<orbit_client_protos::CallstackEvent, 16 * 1024>&
+  GetCallstackEvents() { return callstack_events_; };
+
+  [[nodiscard]] uint32_t GetCallstackEventsSize() {
```
[the remark this answers is not in the document]
Comment (IrinaShkviro) [reply OrbitCore/CallstackData.h]: Done

File: OrbitCore/CallstackData.h [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+
+class CallstackData {
+ public:
+  explicit CallstackData() = default;
+
+  void AddCallstackEvent(orbit_client_protos::CallstackEvent&& callstack_event);
+  void AddUniqueCallStack(CallStack call_stack);
+  void AddUniqueCallStack(std::shared_ptr<CallStack> call_stack);
+
+  [[nodiscard]] BlockChain<orbit_client_protos::CallstackEvent, 16 * 1024>&
```
[the remark this answers is not in the document]
Comment (IrinaShkviro) [reply OrbitCore/CallstackData.h]: Yes, fixed it

File: OrbitCore/CaptureData.h [hunk 9]
```diff
@@ -103,6 +110,23 @@ class CaptureData {
...
+  [[nodiscard]] std::unique_ptr<CallstackData>& GetCallstackData() {
+    return callstack_data_;
+  }
+
+  void SetSelectionCallstackData(
+      std::unique_ptr<CallstackData> callstack_data) {
+    selection_callstack_data_ = std::move(callstack_data);
+  }
+
+  [[nodiscard]] std::unique_ptr<CallstackData>& GetSelectionCallstackData() {
```
[the remark this answers is not in the document]
Comment (IrinaShkviro) [reply OrbitCore/CaptureData.h]: Thanks, made it a bare pointer

File: OrbitCore/SamplingProfiler.cpp [hunk 10]
```diff
@@ -217,24 +200,15 @@ void SamplingProfiler::ProcessSamples() {
...
 
-void SamplingProfiler::ResolveCallstacks() {
-  absl::MutexLock lock(&unique_callstacks_mutex_);
-
-  for (const auto& it : unique_callstacks_) {
-    CallstackID raw_callstack_id = it.first;
-    const std::shared_ptr<CallStack> callstack = it.second;
-    // A "resolved callstack" is a callstack where every address is replaced by
-    // the start address of the function (if known).
+void SamplingProfiler::ResolveCallstacks(CallstackData* callstack_data) {
```
[the remark this answers is not in the document]
Comment (IrinaShkviro) [reply OrbitCore/SamplingProfiler.cpp]: done

File: OrbitGl/TimeGraph.cpp (hunk 1 above)
Comment (IrinaShkviro) [reply OrbitGl/TimeGraph.cpp]: I created a method `AddCallStackFromCallstackData` to fill SelectionCallstackData and introduced variables as well

File: OrbitCore/CallstackData.cpp [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+
+#include "CallstackData.h"
+
+#include "Callstack.h"
+#include "Capture.h"
+#include "SamplingProfiler.h"
+
+using orbit_client_protos::CallstackEvent;
+
+void CallstackData::AddCallstackEvent(CallstackEvent&& callstack_event) {
```
Comment (florian-kuebler) [inline OrbitCore/CallstackData.cpp:13]: Can you add a comment at the declaration of this function, that it assumes that the callstack hash is known?

File: OrbitCore/CallstackData.cpp [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+  CallstackID hash = callstack_event.callstack_hash();
+  CHECK(HasCallStack(hash));
+  callstack_events_.push_back(std::move(callstack_event));
```
Comment (florian-kuebler) [inline OrbitCore/CallstackData.cpp:16]: Any reason why we need a lock below, but not here?

File: OrbitCore/CallstackData.cpp [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+void CallstackData::AddUniqueCallStack(CallStack call_stack) {
+  absl::MutexLock lock(&unique_callstacks_mutex_);
+  CallstackID hash = call_stack.GetHash();
+  unique_callstacks_[hash] = std::make_shared<CallStack>(std::move(call_stack));
+}
+
+void CallstackData::AddCallStackFromCallstackData(
```
Comment (florian-kuebler) [inline OrbitCore/CallstackData.cpp:25]: I guess this is used for selections, where other is the global report? Maybe make this a bit more clear in the naming.
E.g. `other` -> `superset`/`all_callstacks`/`known_callstacks`.

File: OrbitCore/CallstackData.h [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+
+  void AddCallstackEvent(orbit_client_protos::CallstackEvent&& callstack_event);
+  void AddUniqueCallStack(CallStack call_stack);
+  void AddCallStackFromCallstackData(
+      const orbit_client_protos::CallstackEvent& event,
+      const CallstackData& other);
+
+  [[nodiscard]] const BlockChain<orbit_client_protos::CallstackEvent,
+                                 16 * 1024>&
+  GetCallstackEvents() const { return callstack_events_; };
```
Comment (florian-kuebler) [inline OrbitCore/CallstackData.h:29]: For simple getters, the name should be `callstack_events()`.
For setters it would be `set_callstack_events`.
https://google.github.io/styleguide/cppguide.html#Function_Names

File: OrbitCore/CallstackData.h [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+
+  [[nodiscard]] uint32_t GetCallstackEventsSize() const {
+    return callstack_events_.size();
+  };
+
+  const CallStack& GetCallStack(CallstackID callstack_id) const {
+    absl::MutexLock lock(&unique_callstacks_mutex_);
+    return *unique_callstacks_.at(callstack_id);
```
Comment (florian-kuebler) [inline OrbitCore/CallstackData.h:37]: This will fail, if the callstack is not available.
You should either add a comment, or maybe even better, consider using `find` and return a bare pointer or an `outcome`.

File: OrbitCore/CallstackData.h [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+  }
+
+  bool HasCallStack(CallstackID callstack_id) const {
+    absl::MutexLock lock(&unique_callstacks_mutex_);
+    return unique_callstacks_.count(callstack_id) > 0;
+  }
+
+  void ForEachUniqueCallstack(
```
Comment (florian-kuebler) [inline OrbitCore/CallstackData.h:45]: As I come from functional programming, I like these for each methods. However, I had a similar discussion with @dimitry- and I believe returning a const ref to `unique_callstacks_`, should be sufficient. But no need to change this from my side.

File: OrbitCore/CallstackData.h [hunk 17]
```diff
@@ -0,0 +1,82 @@
...
+  void ForEachFrameInCallstack(
+      uint64_t callstack_id,
+      const std::function<void(uint64_t)>& action) const {
+    absl::MutexLock lock(&unique_callstacks_mutex_);
+    for (uint64_t frame : unique_callstacks_.at(callstack_id)->GetFrames()) {
+      action(frame);
+    }
+  }
+
+  void ClearCallstacks() {
```
Comment (florian-kuebler) [inline OrbitCore/CallstackData.h:62]: I think at some point soon, we want to get rid of this method and just ensure, that callstack_data (or better its container), gets reassigned. 
Can you add a TODO to carefully remove this at some point?

File: OrbitCore/CallstackData.h [hunk 18]
```diff
@@ -0,0 +1,82 @@
...
+    absl::MutexLock lock(&unique_callstacks_mutex_);
+    unique_callstacks_.clear();
+    callstack_events_.clear();
+  }
+
+ private:
+  std::shared_ptr<CallStack> GetCallstackPtr(CallstackID callstack_id) const {
+    absl::MutexLock lock(&unique_callstacks_mutex_);
+    return unique_callstacks_.at(callstack_id);
```
Comment (florian-kuebler) [inline OrbitCore/CallstackData.h:71]: As this will fail if not present, please use `find` and return `nullptr` if not present.

File: OrbitCore/CaptureData.h [hunk 19]
```diff
@@ -118,6 +142,10 @@ class CaptureData {
 
   absl::flat_hash_map<uint64_t, orbit_client_protos::FunctionStats>
       functions_stats_;
+
+  std::unique_ptr<CallstackData> callstack_data_;
```
Comment (florian-kuebler) [inline OrbitCore/CaptureData.h:146]: Any reason for those two the be unique_ptrs rather than just members? Just a clarification question.

File: OrbitCore/CaptureData.h [hunk 20]
```diff
@@ -103,6 +110,23 @@ class CaptureData {
...
+  [[nodiscard]] CallstackData* GetCallstackData() {
+    return callstack_data_.get();
+  }
+
+  void SetSelectionCallstackData(
+      std::unique_ptr<CallstackData> callstack_data) {
+    selection_callstack_data_ = std::move(callstack_data);
+  }
+
+  [[nodiscard]] CallstackData* GetSelectionCallstackData() {
```
Comment (florian-kuebler) [inline OrbitCore/CaptureData.h:126]: Can this be const?

File: OrbitCore/CaptureData.h [hunk 21]
```diff
@@ -103,6 +110,23 @@ class CaptureData {
   void UpdateFunctionStats(orbit_client_protos::FunctionInfo* func,
                            const orbit_client_protos::TimerInfo& timer_info);
 
+  void SetCallstackData(std::unique_ptr<CallstackData> callstack_data) {
+    callstack_data_ = std::move(callstack_data);
+  }
+
+  [[nodiscard]] CallstackData* GetCallstackData() {
```
Comment (florian-kuebler) [inline OrbitCore/CaptureData.h:117]: Can this method be const and return a const pointer?

File: OrbitCore/CallstackData.h [hunk 22]
```diff
@@ -0,0 +1,82 @@
...
+  }
+
+  void ClearCallstacks() {
+    absl::MutexLock lock(&unique_callstacks_mutex_);
+    unique_callstacks_.clear();
+    callstack_events_.clear();
+  }
+
+ private:
+  std::shared_ptr<CallStack> GetCallstackPtr(CallstackID callstack_id) const {
```
Comment (florian-kuebler) [inline OrbitCore/CallstackData.h:69]: Can it be also return a const ptr?
Also could you add [[nodiscard]]?

File: OrbitCore/CallstackData.h [hunk 23]
```diff
@@ -0,0 +1,82 @@
...
+  [[nodiscard]] uint32_t GetCallstackEventsSize() const {
+    return callstack_events_.size();
+  };
+
+  const CallStack& GetCallStack(CallstackID callstack_id) const {
+    absl::MutexLock lock(&unique_callstacks_mutex_);
+    return *unique_callstacks_.at(callstack_id);
+  }
+
+  bool HasCallStack(CallstackID callstack_id) const {
```
Comment (florian-kuebler) [inline OrbitCore/CallstackData.h:40]: [[nodiscard]]?

File: OrbitCore/CallstackData.h [hunk 24]
```diff
@@ -0,0 +1,82 @@
...
+
+  [[nodiscard]] const BlockChain<orbit_client_protos::CallstackEvent,
+                                 16 * 1024>&
+  GetCallstackEvents() const { return callstack_events_; };
+
+  [[nodiscard]] uint32_t GetCallstackEventsSize() const {
+    return callstack_events_.size();
+  };
+
+  const CallStack& GetCallStack(CallstackID callstack_id) const {
```
Comment (florian-kuebler) [inline OrbitCore/CallstackData.h:35]: [[nodiscard]]?

File: OrbitCore/CallstackData.h (hunk 14 above)
Comment (florian-kuebler) [reply OrbitCore/CallstackData.h:29]: Also, can you add  [[nodiscard]]

File: OrbitCore/SamplingProfiler.cpp [hunk 25]
```diff
@@ -145,8 +128,8 @@ void SamplingProfiler::ProcessSamples() {
   address_to_module_name_.clear();
 
   // Unique call stacks and per thread data
-  for (const CallstackEvent& callstack : callstack_events_) {
-    if (!HasCallStack(callstack.callstack_hash())) {
+  for (const CallstackEvent& callstack : callstack_data.GetCallstackEvents()) {
+    if (!callstack_data.HasCallStack(callstack.callstack_hash())) {
```
Comment (florian-kuebler) [inline OrbitCore/SamplingProfiler.cpp:132]: This would be a programming error, right? So could you change this into a CHECK here?

File: OrbitGl/App.cpp [hunk 26]
```diff
@@ -123,7 +124,7 @@ void OrbitApp::OnKeyAndString(uint64_t key, std::string str) {
 }
 
 void OrbitApp::OnCallstack(CallStack callstack) {
```
Comment (florian-kuebler) [inline OrbitGl/App.cpp:126]: Can you rename this to something like `OnUniqueCallstack`?

File: OrbitGl/App.cpp [hunk 27]
```diff
@@ -123,7 +124,7 @@ void OrbitApp::OnKeyAndString(uint64_t key, std::string str) {
...
-  Capture::GSamplingProfiler->AddUniqueCallStack(callstack);
+  Capture::capture_data_.GetCallstackData()->AddUniqueCallStack(callstack);
```
Comment (florian-kuebler) [inline OrbitGl/App.cpp:127]: I am thinking about if it would be maybe better to hide those methods to CaptureData, and then call CaptureData->AddUniqueCallstack. What do you think? Could be also in a separate PR, and I am not quite sure about this.

File: OrbitCore/CallstackData.cpp (hunk 12 above)
Comment (IrinaShkviro) [reply OrbitCore/CallstackData.cpp:16]: We use lock only for unique_callstacks_, here we change only callstack_events_

Comment (florian-kuebler) [reply OrbitCore/CallstackData.cpp:16]: But should the callstack_events_ also be guarded by a lock?

File: OrbitCore/CallstackData.h (hunk 14 above)
Comment (IrinaShkviro) [reply OrbitCore/CallstackData.h:29]: I thought that it's not a strict rule (so you may use one of two: GetCallstackEvents() or callstack_events()). For which function do you mean add `[[nodiscard]]`?

Comment (florian-kuebler) [reply OrbitCore/CallstackData.h:29]: I though we wanted to make it more strict. I would add them to all getters here.
Basically, I would add them always, when you can consider the behavior of the program to be incorrect, when calling the method and dropping the return value.
In this case: Calling a getter and not using the value, can only be wrong. Thus, you want the compiler to tell you this.

File: OrbitCore/CallstackData.h (hunk 16 above)
Comment (IrinaShkviro) [reply OrbitCore/CallstackData.h:45]: For doing something with `unique_callstacks_` we should lock the `unique_callstacks_mutex_`, and it means that if we want to do something outside, we should get access to this mutex as well. But I would prefer to keep this mutex private.

File: OrbitCore/CaptureData.h (hunk 19 above)
Comment (florian-kuebler) [reply OrbitCore/CaptureData.h:146]: As discussed offline, just add a comment with the explanation here.

File: OrbitGl/App.cpp (hunk 27 above)
Comment (IrinaShkviro) [reply OrbitGl/App.cpp]: Added these methods to CaptureData.

File: OrbitGl/App.cpp (hunk 26 above)
Comment (IrinaShkviro) [reply OrbitGl/App.cpp]: Done

File: OrbitCore/SamplingProfiler.cpp (hunk 25 above)
Comment (IrinaShkviro) [reply OrbitCore/SamplingProfiler.cpp]: Yes, replaced with CHECK

File: OrbitCore/CallstackData.h (hunk 24 above)
Comment (IrinaShkviro) [reply OrbitCore/CallstackData.h]: Done

File: OrbitCore/CallstackData.h (hunk 23 above)
Comment (IrinaShkviro) [reply OrbitCore/CallstackData.h]: Done

File: OrbitCore/CallstackData.h (hunk 22 above)
Comment (IrinaShkviro) [reply OrbitCore/CallstackData.h]: Let's keep it a shared_ptr for now: I want to try to get rid of the shared_ptr here, and in this case we should duplicate CallStack data. So, before doing that change I want to double check that I won't break/slow down something.

File: OrbitCore/CaptureData.h (hunk 20 above)
Comment (IrinaShkviro) [reply OrbitCore/CaptureData.h]: Done

File: OrbitCore/CaptureData.h (hunk 19 above)
Comment (IrinaShkviro) [reply OrbitCore/CaptureData.h:163]: Done

File: OrbitCore/CallstackData.h (hunk 18 above)
Comment (IrinaShkviro) [reply OrbitCore/CallstackData.h]: Added `find` and return const pointer here and in GetCallStack function.

File: OrbitCore/CallstackData.h (hunk 17 above)
Comment (IrinaShkviro) [reply OrbitCore/CallstackData.h]: There is just one place where we use it (in CaptureSerializer), and it doesn't make any sense to keep it here as we always create a new capture_data there, so I've removed it.

File: OrbitCore/CallstackData.h (hunk 15 above)
Comment (IrinaShkviro) [reply OrbitCore/CallstackData.h]: Done

File: OrbitCore/CallstackData.h (hunk 14 above)
Comment (IrinaShkviro) [reply OrbitCore/CallstackData.h]: Renamed the function

File: OrbitCore/CallstackData.cpp (hunk 13 above)
Comment (IrinaShkviro) [reply OrbitCore/CallstackData.cpp]: Renamed the argument and the function.

File: OrbitCore/CallstackData.cpp (hunk 11 above)
Comment (IrinaShkviro) [reply OrbitCore/CallstackData.cpp:13]: Done

File: OrbitCore/CallstackData.cpp (hunk 12 above)
Comment (IrinaShkviro) [reply OrbitCore/CallstackData.cpp:16]: Probably yes... But we should understand if it implies on performance...

**Comment by IrinaShkviro** [comment]
I was typing too slow :)

File: OrbitCore/CallstackData.cpp (hunk 11 above)
Comment (florian-kuebler) [reply OrbitCore/CallstackData.cpp:13]: I think we prefer value arguments for sinks, and move them explicitly on the call side, such that you can see on the caller, that the object becomes invalid after the call.

File: OrbitCore/CallstackData.cpp [hunk 28]
```diff
@@ -0,0 +1,40 @@
...
+
+using orbit_client_protos::CallstackEvent;
+
+void CallstackData::AddCallstackEvent(CallstackEvent&& callstack_event) {
+  CallstackID hash = callstack_event.callstack_hash();
+  CHECK(HasCallStack(hash));
+  callstack_events_.push_back(std::move(callstack_event));
+}
+
+void CallstackData::AddUniqueCallStack(CallStack&& call_stack) {
```
Comment (florian-kuebler) [inline OrbitCore/CallstackData.cpp:19]: Same as above. Should probably value argument and move on call.

File: OrbitCore/CaptureData.h [hunk 29]
```diff
@@ -103,6 +110,38 @@ class CaptureData {
...
 
+  [[nodiscard]] const CallstackData* GetCallstackData() const {
+    return callstack_data_.get();
+  };
+
+  void AddUniqueCallStack(CallStack&& call_stack) {
+    callstack_data_->AddUniqueCallStack(std::move(call_stack));
+  }
+
+  void AddCallstackEvent(
```
Comment (florian-kuebler) [inline OrbitCore/CaptureData.h:121]: As above, probably we want value args here.

File: OrbitCore/CaptureData.h [hunk 30]
```diff
@@ -103,6 +110,38 @@ class CaptureData {
...
+      orbit_client_protos::CallstackEvent&& callstack_event) {
+    callstack_data_->AddCallstackEvent(std::move(callstack_event));
+  }
+
+  [[nodiscard]] const CallstackData* GetSelectionCallstackData() const {
+    return selection_callstack_data_.get();
+  };
+
+  void AddSelectionCallStack(CallStack&& call_stack) {
```
Comment (florian-kuebler) [inline OrbitCore/CaptureData.h:130]: And here.

File: OrbitCore/CaptureData.h [hunk 31]
```diff
@@ -103,6 +110,38 @@ class CaptureData {
...
+    selection_callstack_data_->AddUniqueCallStack(std::move(call_stack));
+  }
+
+  void AddSelectionCallstackEvent(
```
Comment (florian-kuebler) [inline OrbitCore/CaptureData.h:134]: also here.

File: OrbitCore/CallstackData.cpp (hunk 11 above)
Comment (IrinaShkviro) [reply OrbitCore/CallstackData.cpp]: Done

File: OrbitCore/CallstackData.cpp (hunk 28 above)
Comment (IrinaShkviro) [reply OrbitCore/CallstackData.cpp]: Done

File: OrbitCore/CaptureData.h (hunk 30 above)
Comment (IrinaShkviro) [reply OrbitCore/CaptureData.h]: Done

File: OrbitCore/CaptureData.h (hunk 31 above)
Comment (IrinaShkviro) [reply OrbitCore/CaptureData.h:133]: Done

File: OrbitCore/CaptureData.h (hunk 29 above)
Comment (IrinaShkviro) [reply OrbitCore/CaptureData.h]: Done
